src/main/java/com/shop/model/User.java

```diff
@@ -18,22 +18,41 @@
 @Table(name = "users")
 public class User implements UserDetails {
 
-
     @Id
     @GeneratedValue
     @Column(name = "user_id")
     private long id;
 
     @NotBlank
+    @Column(unique = true)
     private String username;
 
     @JsonIgnore
     @NotBlank
     private String password;
 
     @NotBlank
+    @Column(unique = true)
     private String email;
 
+    @NotBlank
```

Comment:
You made all fields required. I think we only need to know mail and password. User can provide rest of the data later during shipping chose. Registration form is too long IMO.
Also u can extract dto layer to separated file and make dto to domain converter in user serivce.
Another thing is that you could provide better validation. Now i can register with email: dupaxd, that's not good.